deepchem/data/data_loader.py

```diff
@@ -674,13 +725,13 @@ def _get_shards(self, input_files, shard_size):
 
   def _featurize_shard(self, shard):
     """Featurizes a shard of an input dataframe."""
-    logger.info("Currently featurizing feature_type: %s" %
-                self.featurizer.__class__.__name__)
-    return _featurize_mol_df(
-        shard,
-        self.featurizer,
-        field=self.mol_field,
-        log_every_n=self.log_every_n)
+    features = [elt for elt in self.featurizer(shard[self.mol_field])]
+    valid_inds = np.array(
+        [1 if np.array(elt).size > 0 else 0 for elt in features], dtype=bool)
+    features = [
+        elt for (is_valid, elt) in zip(valid_inds, features) if is_valid
```

Comment:
Same comment as above for these list comprehensions.

Reply:
I think for now, I'll keep the list comprehensions, but if this turns out to be a performance issue will replace!